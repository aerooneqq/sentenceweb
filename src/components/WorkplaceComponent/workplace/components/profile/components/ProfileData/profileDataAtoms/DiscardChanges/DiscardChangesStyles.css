#discardChangesCont{ 
    width: 90%;
    height: auto;

    font-family: Arial; 
    font-size: 15px;
    color: gray; 
    
    margin-left: 10%;
    margin-top: 10px;
    margin-bottom: 10px;

    cursor: default;
}

#discardChangesCont:hover{ 
    text-decoration: underline;
}