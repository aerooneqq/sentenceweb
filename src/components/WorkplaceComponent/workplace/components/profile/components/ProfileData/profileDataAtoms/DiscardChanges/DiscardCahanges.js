import React, {Component} from "react"

//Styles
import "./DiscardChangesStyles.css"

/**
 * PROPS LIST:
 * 1) discardChangesInProfileData - the function which discards the changes in profile data.
 */
export default class DiscardChanges extends Component{ 
    constructor(props){ 
        super(props);

        this.handleDiscardChangesBtnClick = this.handleDiscardChangesBtnClick.bind(this);
    }

    handleDiscardChangesBtnClick(){ 
        this.props.discardChangesInProfileData();
    }

    render(){ 
        return(
            <div id = "discardChangesCont" onClick={this.handleDiscardChangesBtnClick}>
                Discard changes
            </div>
        );
    }
}