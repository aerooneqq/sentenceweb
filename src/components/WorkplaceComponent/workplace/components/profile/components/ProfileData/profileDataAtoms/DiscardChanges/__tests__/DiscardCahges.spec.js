import React from "react";
import {shallow} from "enzyme";
import {shallowToJson} from "enzyme-to-json";
import DiscardChanges from "../DiscardCahanges";

const spy = jest.fn(() => {})
const props = { 
    discardChangesInProfileData: spy
}

describe("DiscardChanges", () => {
    it("should render correctly", () => { 
        expect(shallowToJson(shallow(<DiscardChanges />))).toMatchSnapshot();
    });

    it("should call the discardChangesInProfileData method from props when clicked", () => { 
        const discardChanges = shallow(<DiscardChanges {...props}/>);

        discardChanges.find("div").simulate("click");

        expect(spy).toBeCalled();
    });
});