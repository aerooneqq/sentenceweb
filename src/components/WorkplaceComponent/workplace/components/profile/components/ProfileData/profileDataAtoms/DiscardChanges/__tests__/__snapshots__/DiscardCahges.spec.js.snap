// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DiscardChanges should render correctly 1`] = `
<div
  id="discardChangesCont"
  onClick={[Function]}
>
  Discard changes
</div>
`;
